sample/NodeResume/HttpTrigger/index.js

```diff
@@ -0,0 +1,14 @@
+﻿function delay(ms) {
```

Comment:
Should we also test with an in-proc .NET Function and use/have an example of using the Cancellation Token? Not sure when the token would get raised to the function while the host is in drain mode, would be great to know so I can document. And then what happens when resume is called - is there a possibility where the Cancellation Token is raised AND resume is called?

Reply:
@soninaren

Reply:
I mean is there a possibility of a cancellation token being propagated to a running function because Drain Mode was called, and then Resume is called while the function code is dealing with the cancellation token? Likely an edge scenario, just wondering if there's any concerns or testing to do for it.

Reply:
When drain is initiated for a host the cancellation tokens passed into any in progress invocations will be cancelled. When resume is called later, we do a restart - the existing (draining) host is "orphaned" and a new host instance is started. The orphaned (draining) instance continues to finish up in the background - once it's listeners are completely stopped and all outstanding invocations are complete, it's disposed. Thus these are separate host instances, so the answer to Thiago's initial question is no - once cancellation for a particular function invocation is triggered, that host is stopping/draining/orphaned and won't be used anymore. I.e. once a JobHost/ScriptHost transitions to draining/orphaned it stays in that state.

Reply:
@mathewc Thanks for confirming!

@nzthiago I can add a new test using cancellation in a new PR for drain mode. I tested something like this and it works as expected, this might help with your drain mode documentation?

```csharp
public static async Task<IActionResult> Run(HttpRequest req, ILogger log, CancellationToken token)
{
    log.LogInformation("C# HTTP trigger cancellation function processing a request.");

    try
    {
        await Task.Delay(5000, token);
        return new OkResult();

    }
    catch (OperationCanceledException)
    {
        log.LogInformation("Function invocation cancelled.");
        return new NotFoundResult();
    }
    catch (Exception e)
    {
        return new StatusCodeResult(500);
    }
}
```

Reply:
@mathewc thank you, makes sense! I forgot about that recycle of JobHost 👍. @liliankasem yes that'd be great to have a recommended way to handle the token in a separate PR as part of a test for drain mode, and I can use it for the documentation update. Appreciate it